rewrite-java/src/main/java/org/openrewrite/java/ChangeType.java

```diff
@@ -527,6 +553,23 @@ private boolean hasNoConflictingImport(@Nullable JavaSourceFile sf) {
         }
     }
 
+    @Value
+    @EqualsAndHashCode(callSuper = false)
+    private static class ReferenceChangeTypeVisitor extends TreeVisitor<Tree, ExecutionContext> {
+        Map<Tree, Reference> matches;
+        Reference.Renamer renamer;
+        String newFullyQualifiedName;
+
+        @Override
+        public @Nullable Tree preVisit(@Nullable Tree tree, ExecutionContext ctx) {
```

Comment:
One more.

Reply:
In this case I probably didn't look carefully enough. I suspect here we need to visit the entire tree.